nx/test/nx/defn_test.exs

```diff
@@ -1319,6 +1319,18 @@ defmodule Nx.DefnTest do
       end
     end
 
+    test "undefined local function" do
+      assert_raise CompileError, ~r"#{location(+5)}: undefined function do_add/2 \(there is no such import\)", fn ->
+        defmodule Sample do
+          import Nx.Defn
+
+          defn add(a, b) do
+            do_add(a, b)
+          end
```

Comment:
What happens if you call a function which is defined but isnnot a defn?

Reply:
Good call, it will be treated as undefined, I will add another error message for it.